src/Docker/Configuration/Container.php

```diff
@@ -148,6 +148,74 @@ public function getConfigTreeBuilder()
                 ->end()
             ->end()
         ->end();
+
+        // artifacts
+        $root->children()
+            ->arrayNode('artifacts')
+            ->validate()
+                ->ifTrue(function ($v) {
+                    $enabledFilters = array_filter($v, function ($filter) {
+                        return isset($filter['enabled']) && $filter['enabled'] === true;
+                    });
+                    return count($enabledFilters) > 1;
+                })
+                ->thenInvalid('Multiple artifacts types, enable only one of "custom", "orchestration" or "runs".')
+            ->end()
+            ->children()
+                ->arrayNode('runs')
+                    ->validate()
+                        ->ifTrue(function ($v) {
+                            if ($v['enabled'] === true) {
+                                if (!isset($v['filter']['date_since']) && !isset($v['filter']['limit'])) {
+                                    return true;
+                                }
+                            }
+                            return false;
+                        })
+                        ->thenInvalid('At least one of "date_since" or "limit" parameters must be defined.')
+                    ->end()
+                    ->children()
+                        ->booleanNode('enabled')->defaultFalse()->end()
+                        ->arrayNode('filter')
+                            ->children()
+                                ->scalarNode('date_since')->end()
+                                ->integerNode('limit')->end()
+                            ->end()
+                        ->end()
+                    ->end()
+                ->end()
+                ->arrayNode('custom')
+                    ->validate()
+                        ->ifTrue(function ($v) {
+                            if ($v['enabled'] === true) {
+                                if (!isset($v['filter']['component_id']) && !isset($v['filter']['config_id']) && !isset($v['filter']['branch_id'])) {
+                                    return true;
+                                }
+                            }
+                            return false;
+                        })
+                        ->thenInvalid('At least one of "component_id", "config_id" or "branch_id" parameters must be defined.')
+                    ->end()
+                    ->children()
+                        ->booleanNode('enabled')->defaultFalse()->end()
+                        ->arrayNode('filter')
+                            ->children()
+                                ->scalarNode('component_id')->end()
```

Comment:
V zadání je to jako camel case, ale většina ostatních věcí v definici je snake case. Proto ty "_"

Reply:
snake_case lepsi v configu urcite